src/screens/inventory.py

```diff
@@ -0,0 +1,144 @@
+
+    def _inventory_part_btn_setup(self, button_size: tuple[int, int]):
+        # Portion of the menu to allow the player to see
+        # how many of each resource they gathered,
+        # and possibly assign them as their current seed
+        # (if the selected one is a seed).
+        generic_rect = pygame.Rect((0, 0), button_size)
+        available_width_for_btns = (self.size[0] * 2 // 3)
+        btns_per_line = available_width_for_btns // button_size[0]
+        x_spacing = (available_width_for_btns % button_size[0]) // max(1, btns_per_line - 1)
+        for button_no, (ir, count) in enumerate(self._inventory.items()):
+            match ir:
+                case InventoryResource.APPLE:
+                    btn_name = "apple"
+                    img = self.obj_frames["apple"]
+                case _:
+                    btn_name = self._IR_TO_OVERLAY_IMG[ir]
+                    img = self.overlay_frames[btn_name]
+            calc_rect = img.get_frect(center=(32, 32))
+            calc_img = pygame.Surface((64, 64), pygame.SRCALPHA)
+            amount = self.font.render(str(count), False, "black")
+            blit_list = (
+                (img, calc_rect),
+                (amount, amount.get_frect(bottomright=(64, 64)))
+            )
+            calc_img.fblits(blit_list)  # faster than doing two separate blits
+            row, column = divmod(button_no, btns_per_line)
+            btn_rect = generic_rect.copy()
+            btn_rect.x = _LEFT_MARGIN + button_size[0]*column + x_spacing * column
+            btn_rect.y = _TOP_MARGIN + (button_size[1] + _SPACING_BETWEEN_ROWS) * row
+            yield _IMButton(calc_img, btn_rect, btn_name)
+
+    def _ft_btn_setup(self, button_size: tuple[int, int]):
+        # Portion of the menu to allow the player to select their current tool.
+        rect = pygame.Rect((0, 0), button_size)
+        rect.centerx = (self.rect.width / 2)
+        for index, tool in enumerate(self._av_tools):
+            img = self.overlay_frames[tool]
+            calc_img = pygame.Surface((64, 64), pygame.SRCALPHA)
+            calc_img.blit(img, img.get_frect(center=(32, 32)))
+            btn_rect = rect.copy()
+            btn_rect.y = _TOP_MARGIN + (button_size[1] + _SPACING_BETWEEN_ROWS) * index
+            yield _IMButton(calc_img, btn_rect, tool)
+
+    def _special_btn_setup(self):
+        # TODO: this part requires separate icons for the goggles, the hat and all special items.
+        # Part of the menu for items such as the goggles, the hat, etc.
+        pass
+
+    def button_action(self, *args, **kwargs):
+        # TODO: add the different assign actions.
+        pass
+
+    def button_setup(self):
+        self._inv_buttons.extend(self._inventory_part_btn_setup(_BUTTON_SIZE))
+        self.buttons.extend(self._inv_buttons)
+        self._ft_buttons.extend(self._ft_btn_setup(_BUTTON_SIZE))
+        self.buttons.extend(self._ft_buttons)
+
+    def draw_title(self):
```

Comment:
i think we might have a update(self) method and draw(self) method too
even if inside they have just self.draw_title and update_display respectively

Reply:
No need. Did you read AbstractMenu?

Reply:
there is no update_display in AbstractMenu, but you're right on the draw method tho

Reply:
update_display is different, I should have commented that. update_display reruns the button setup, but only changed the buttons for resources

Reply:
It does that so we don’t end up with a memory leak, as we delete the old buttons and then make new ones

Reply:
> update_display is different, I should have commented that. update_display reruns the button setup, but only changed the buttons for resources

maybe changing the method name to refresh_buttons_content or something else


> It does that so we don’t end up with a memory leak, as we delete the old buttons and then make new ones

i mean i thought it had to be called by the update method but it's not an update method its not used every while game loop